src/app/leverage/provider.tsx

```diff
@@ -235,12 +238,21 @@ export function LeverageProvider(props: { children: any }) {
   }, [chainId, indexToken])
 
   useEffect(() => {
-    if (!publicClient || inputToken === null || outputToken === null) return
+    async function fetchCosts() {
+      const carryCosts = await fetchCarryCosts()
+      setCarryCosts(carryCosts)
+    }
+
+    fetchCosts()
+  }, [])
+
+  useEffect(() => {
+    if (inputToken === null || outputToken === null) return
 
-    const jsonRpcProvider = publicClientToProvider(publicClient)
     const inputOutputToken = isMinting ? outputToken : inputToken
-    fetchCostOfCarry(jsonRpcProvider, inputOutputToken, setCostOfCarry)
-  }, [publicClient, isMinting, inputToken, outputToken])
+    if (carryCosts)
+      setCostOfCarry(carryCosts[inputOutputToken.symbol.toLowerCase()] ?? null)
```

Comment:
@janndriessen one item we have now as part of this UI that we didn't before is ETH2x on mainnet - does the cost of carry still apply there?

Reply:
Haven't heard of it there before, so I guess can be omitted for now. Best to check with caf when she's back.